Sources/FlowLayoutStack.swift

```diff
@@ -0,0 +1,122 @@
+    private var itemSpacing: CGFloat {
+        spacing
+    }
+    
+    public init(
+        itemSpacing: CGFloat = 0.0,
+        lineSpacing: CGFloat = 0.0,
+        width: CGFloat? = nil,
+        thingsToStack: [Stackable]
+    ) {
+        self.spacing = itemSpacing
+        self.lineSpacing = lineSpacing
+        self.layoutMargins = .zero
+        self.width = width
+        self.thingsToStack = thingsToStack
+    }
+    
+    public convenience init(
+        itemSpacing: CGFloat = 0.0,
+        lineSpacing: CGFloat = 0.0,
+        width: CGFloat? = nil,
+        thingsToStack: () -> [Stackable]
+    ) {
+        self.init(
+            itemSpacing: itemSpacing,
+            lineSpacing: lineSpacing,
+            width: width,
+            thingsToStack: thingsToStack()
+        )
+    }
+    
+    open func framesForLayout(_ width: CGFloat, origin: CGPoint) -> [CGRect] {
+        var frames: [CGRect] = []
+        var currentY = origin.y
+        var currentX = origin.x
+
+        func moveToNextRow() {
+            currentX = 0
+            currentY = frames.reduce(0) { result, rect in
+                max(
+                    result,
+                    rect.origin.y + rect.size.height
+                )
+            } + lineSpacing
+        }
+        
+        for stackable in self.visibleThingsToStack() {
+            let stackableWidth = getWidth(
+                for: stackable,
+                width: width
+            )
+                            
+            if currentX + stackableWidth > width {
+                moveToNextRow()
+            }
+            
+            if let stack = stackable as? Stack {
+                frames.append(
+                    contentsOf: stack.framesForLayout(
+                        (width - currentX),
+                        origin: .init(
+                            x: currentX,
+                            y: currentY
+                        )
+                    )
+                )
+            } else if let item = stackable as? StackableItem {
+                frames.append(
+                    .init(
+                        x: currentX,
+                        y: currentY,
+                        width: stackableWidth,
+                        height: item.heightForWidth(stackableWidth)
+                    )
+                )
+            }
+
+            currentX += stackableWidth + itemSpacing
+        }
+
+        return frames
+    }
+    
+    private func getWidth(for stackable: Stackable, width: CGFloat) -> CGFloat {
+        return if let stackableItem = stackable as? StackableItem {
+            min(
+                stackableItem.intrinsicContentSize.width,
+                width
+            )
+        } else if let fixedSizeStack = stackable as? Stack, let stackWidth = fixedSizeStack.width {
+            stackWidth
+        } else {
+           stackable.intrinsicContentSize.width
+        }
+    }
+    
+    public var intrinsicContentSize: CGSize {
```

Comment:
This is part of the contract of Stack.  I have leveraged the same HStack `intrinsicContentSize` logic because given there is now bound width it can size itself as wide as it wants and will not wrap.

Reply:
what do you think of subclassing from HStack?

Reply:
Yep makes sense it is a kind of special HStack 👍 ... done